polyfill/lib/date.mjs

```diff
@@ -53,24 +53,29 @@ export class Date {
     return ES.LeapYear(GetSlot(this, YEAR));
   }
   with(dateLike = {}, disambiguation = 'constrain') {
+    if (!ES.ValidPropertyBag(dateLike, [ 'year', 'month', 'day' ])) {
+      throw new RangeError('invalid date-like');
+    }
     const { year = GetSlot(this, YEAR), month = GetSlot(this, MONTH), day = GetSlot(this, DAY) } = dateLike;
     return new Date(year, month, day, disambiguation);
   }
   plus(durationLike = {}, disambiguation = 'constrain') {
     const duration = ES.CastDuration(durationLike);
-    let { year, month, day } = this;
-    let { years, months, days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = duration;
-    if (hours || minutes || seconds || milliseconds || microseconds || nanoseconds)
+    if (!ES.ValidPropertyBag(duration, [], [ 'hours', 'minutes', 'seconds', 'milliseconds', 'microseconds', 'nanoseconds' ])) {
```

Comment:
I don't understand the value of this check. In any case, it should use the internal slots, not the (possibly tampered-with) getters.

Reply:
The reason is that `date.plus({ seconds: 60 })` does not make sense. So the duration you pass in is not allowed to have a positive `seconds` value.

My mistake in this PR (**will change**): I combined the duration checking and the validity checking into 1 operation.